sys/net/gnrc/network_layer/ndp/gnrc_ndp.c

```diff
@@ -260,6 +274,172 @@ void gnrc_ndp_nbr_adv_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
     return;
 }
 
+#if defined(MODULE_GNRC_NDP_ROUTER) || defined(MODULE_GNRC_SIXLOWPAN_ND_ROUTER)
+void gnrc_ndp_rtr_sol_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
+                             ipv6_hdr_t *ipv6, ndp_rtr_sol_t *rtr_sol,
+                             size_t icmpv6_size)
+{
+    gnrc_ipv6_netif_t *if_entry = gnrc_ipv6_netif_get(iface);
+    assert(if_entry != NULL);
+    if (if_entry->flags & GNRC_IPV6_NETIF_FLAGS_ROUTER) {
+        int sicmpv6_size = (int)icmpv6_size, l2src_len = 0;
+        uint8_t l2src[GNRC_IPV6_NC_L2_ADDR_MAX];
+        uint16_t opt_offset = 0;
+        uint8_t *buf = (uint8_t *)(rtr_sol + 1);
+        /* check validity */
+        if ((ipv6->hl != 255) || (rtr_sol->code != 0) ||
+            (icmpv6_size < sizeof(ndp_rtr_sol_t))) {
+            DEBUG("ndp: router solicitation was invalid\n");
+            return;
+        }
+        sicmpv6_size -= sizeof(ndp_rtr_sol_t);
+        while (sicmpv6_size > 0) {
+            ndp_opt_t *opt = (ndp_opt_t *)(buf + opt_offset);
+
+            switch (opt->type) {
+                case NDP_OPT_SL2A:
+                    if ((l2src_len = gnrc_ndp_internal_sl2a_opt_handle(pkt, ipv6, rtr_sol->type, opt,
+                                                                       l2src)) < 0) {
+                        if (l2src_len != -ENOTSUP) {
+                            /* invalid source link-layer address option */
+                            return;
+                        }
+                    }
+                    break;
+
+                default:
+                    /* silently discard all other options */
+                    break;
+            }
```

Comment:
Not by 6LoWPAN-ND, but if you want to know you can research at http://www.iana.org/assignments/icmpv6-parameters/icmpv6-parameters.xhtml#icmpv6-parameters-6 what other options can be contained in an RS :D